app/src/main/java/de/danoeh/antennapod/menuhandler/FeedItemMenuHandler.java

```diff
@@ -154,24 +155,24 @@ public static boolean onMenuItemClicked(@NonNull Fragment fragment, int menuItem
             if (GpodnetPreferences.loggedIn()) {
                 FeedMedia media = selectedItem.getMedia();
                 // not all items have media, Gpodder only cares about those that do
-                if (media != null) {
+                if (media != null && SynchronizationSettings.isSynchronizationProviderActive()) {
                     EpisodeAction actionPlay = new EpisodeAction.Builder(selectedItem, EpisodeAction.PLAY)
                             .currentTimestamp()
                             .started(media.getDuration() / 1000)
                             .position(media.getDuration() / 1000)
                             .total(media.getDuration() / 1000)
                             .build();
-                    SyncService.enqueueEpisodeAction(context, actionPlay);
+                    LockingQueueWriter.enqueueEpisodeAction(context, actionPlay);
```

Comment:
TL;DR; i'd rather not
I moved it out of the QueueWriter because i expect the queue to be written when i call its methods. Internally (inside the method) not executing the described functionality (enqueueing stuff) because of some condition is bad practice.

Also speaking of seperation of concerns: the QueueWriter should not know about active state of providers. Its only purpose is to write to the queue.

Reply:
I agree that the code structure is nicer when checking outside of it. What worries me is that one always has to keep in mind to check this. I mean, sure, in this PR alone it is easy. You can just let AndroidStudio find all usages. If someone new to AntennaPod proposes a PR, though, it is easy to miss in new code. When the API does not allow to do the wrong thing in the first place, it is a lot more robust.

Even though I do pretty detailed reviews, I currently need to fix a pretty high number of bugs introduced in recent PRs. Having to remember doing this check is another possible source of future problems that I would rather avoid. I want to get back to adding/improving features myself instead of only having to fix stuff that people break in their PRs.

Reply:
The API clearly says: enqueue stuff. If enqueueing might not actually happen it must be made explicit by naming or...

What about an event system? We fire enqueue events and a listener is only there if the we have an active synchronization provider.

Reply:
I'm not 100% sure how you plan to do that but I like the idea. It helps to remove the cyclic dependency *Database » Sync » Database*, untangling it to *Sync » Database » Event*

Reply:
I think we have done enough refactoring in the sync module. Implementing an event/subscriber bus goes way beyond what this PR wants to contribute. I moved the condition into the methods and renamed them accordingly.